### ISSUE TITLE
`Fetching latest Viceroy release` makes running in a container unreliable

### ISSUE THREAD

**Issue by twhid** [fastly/cli#828]
**Version**
```
Fastly CLI version v6.0.6 (f10a150)
Built with go version go1.20 linux/amd64
```

**What happened**

TL:DR<br>
Is there a way to avoid (or make more reliable) `Fetching latest Viceroy release...` when running `compute serve` ? It sometimes fails, which is causing issues when running the test server in a container.

====

I'm running tests with Fastly CLI in a container using this [Dockerfile](https://github.com/fastly/cli/blob/main/Dockerfile-node).

Using a command like this:
`docker run --rm -v $PWD/app -p 7676:7676 fastly/image/name compute serve --skip-build --addr 0.0.0.0:7676`

And I see this output:
```bash
Fastly API token not provided
Fastly API endpoint: https://api.fastly.com

Checking latest Viceroy release...

Fetching latest Viceroy release...

Running local server...
Wasm file: bin/main.wasm
Manifest: /app/fastly.toml


2023-02-15T16:41:05.339183Z  INFO checking if backend 'web_host' is up

2023-02-15T16:41:05.394217Z  INFO backend 'web_host' is up
2023-02-15T16:41:05.394287Z  INFO checking if backend 'service_host' is up

2023-02-15T16:41:05.435437Z  INFO backend 'service_host' is up

2023-02-15T16:41:05.436662Z  INFO Listening on http://0.0.0.0:7676
# my test output follows
```

But, sometimes, after the `Fetching latest Viceroy release...` I see this output (and the process exits):
```
Test server failed to start with error: 
ERROR: error downloading latest Viceroy release: failed to download release asset: 
context deadline exceeded (Client.Timeout or context cancellation while reading body).
```

Is there a way to skip this fetch? Or does anyone have suggestions on how to make it more reliable?

**Comment by Integralist** [comment]
Hi @twhid 

Thanks for opening this issue. Debugging why the network timed out within a Docker container can be tricky.

Are you seeing this issue outside of Docker?

A way to side-step the issue, if you're sure you have at least a working version of Viceroy installed, would be to update the TTL in the CLI's configuration:

```shell
$ fastly config | grep -i -A 3 viceroy
[viceroy]
last_checked = "2023-02-15T10:19:40Z"
latest_version = "0.3.5"
ttl = "24h"
```

You can find the location by passing the `--location` flag:

```shell
$ fastly config --location
/Users/integralist/Library/Application Support/fastly/config.toml
```

**Comment by twhid** [comment]
It doesn't appear there's a way via the cli to update the config or trigger a viceroy update which would update the config.

**Comment by Integralist** [comment]
👋🏻 

Correct, there are no CLI commands for modifying that configuration file as it's intended to be an internal file and shouldn't really be modified by a user.

It might be best to clarify how those values are currently set...

Every time you run `compute serve` the CLI will check the config for those Viceroy values. 

So the first time you run the CLI (which I guess from Docker's perspective is _every_ time you run the CLI) there will be no values set and so the CLI looks up the latest Viceroy version on GitHub. Once it gets those details it will update the config.

Now in your case the config is likely refreshed every time so the only way I can think to workaround how Docker works would be to mount that config file into the running Docker container so that it has a prior value (or in your case just a default value for each of the viceroy fields).

That said, I've been discussing this issue internally with the team and the option of implementing a `--offline` flag is likely something we'll add to help with scenarios like Docker/CI and then your own process workflow would need to ensure Viceroy was installed.

**Comment by Integralist** [comment]
👋🏻 @twhid so following on from my comments a few weeks ago...

You'll need [Viceroy](https://github.com/fastly/viceroy) installed for `compute serve` to run successfully. 

That means you'll ultimately need to have your CI pipeline install Viceroy and then the following PR should address the CLI side of the issue: https://github.com/fastly/cli/pull/859

**Comment by Integralist** [comment]
I'll hopefully have a new CLI release out by the end of the week.
